runtime/src/qio/sys.c

```diff
@@ -1416,7 +1426,12 @@ err_t sys_connect(fd_t sockfd, const sys_sockaddr_t* addr)
 
   STARTING_SLOW_SYSCALL;
 
+#ifdef QTHREAD_VERSION
+  got = qt_connect(sockfd, (const struct sockaddr*) & addr->addr, addr->len);
+#else
```

Comment:
@mppf @krishnadey30 there seems to be an issue here with `qt_connect`. It isn't returning an error even when the port has no service listening on it where we call it to connect. Normal connect returns a transport fail error, whereas qt_connect returns a zero

Reply:
Can you try to make a C reproducer and post it to the threads issue tracker here: https://github.com/Qthreads/qthreads/issues ? Additionally it would be good if you can check that you can get errno from a failed connect call.

Reply:
Updated this to use `connect` as per discussion is last meeting